server/sonar-server/src/test/java/org/sonar/server/qualitygate/ws/UpdateConditionActionTest.java

```diff
@@ -69,67 +66,78 @@
   private TestDefaultOrganizationProvider defaultOrganizationProvider = TestDefaultOrganizationProvider.from(db);
   private DbClient dbClient = db.getDbClient();
   private DbSession dbSession = db.getSession();
-  private QualityGateDbTester qualityGateDbTester = new QualityGateDbTester(db);
   private UpdateConditionAction underTest = new UpdateConditionAction(userSession, dbClient, new QualityGateConditionsUpdater(dbClient), defaultOrganizationProvider);
-  private QualityGateDto qualityGateDto;
-  private QualityGateConditionDto conditionDto;
-  private MetricDto coverageMetricDto = newMetricDto()
-    .setKey("coverage")
-    .setShortName("Coverage")
-    .setValueType(PERCENT.name())
-    .setHidden(false);
-
-  WsActionTester ws = new WsActionTester(underTest);
-
-  @Before
-  public void setUp() throws Exception {
-    MetricDto metricDto = dbClient.metricDao().insert(dbSession, coverageMetricDto);
-    qualityGateDto = qualityGateDbTester.insertQualityGate();
-    conditionDto = new QualityGateConditionDto().setQualityGateId(qualityGateDto.getId())
-      .setMetricId(metricDto.getId())
-      .setOperator("GT")
-      .setWarningThreshold(null)
-      .setErrorThreshold("80")
-      .setPeriod(1);
-    dbClient.gateConditionDao().insert(conditionDto, dbSession);
-    dbSession.commit();
-  }
+
+  private WsActionTester ws = new WsActionTester(underTest);
 
   @Test
   public void update_warning_condition() throws Exception {
     logInAsQualityGateAdmin();
```

Comment:
Maybe it's valuable to factorize : 
```java
    logInAsQualityGateAdmin();
    QualityGateDto qualityGate = db.qualityGates().insertQualityGate();
    MetricDto metric = db.measures().insertMetric(m -> m.setValueType(INT.name()).setHidden(false));
```

Reply:
IMO, we need the quality gate, and we need the metric to ba able to create a condition.
So I would not factorise more.